pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go

```diff
@@ -52,6 +52,18 @@ type Reconciler struct {
 func (r *Reconciler) ReconcileKind(ctx context.Context, imc *v1.InMemoryChannel) reconciler.Event {
 	logging.FromContext(ctx).Infow("Reconciling", zap.Any("InMemoryChannel", imc))
 
+	// If the IMC has been deleted just make sure the handler is removed.
+	if !imc.GetDeletionTimestamp().IsZero() {
+		if imc.Status.Address != nil &&
+			imc.Status.Address.URL != nil {
+			if hostName := imc.Status.Address.URL.Host; hostName != "" {
+				logging.FromContext(ctx).Info("Removing dispatcher")
+				r.multiChannelMessageHandler.DeleteChannelHandler(hostName)
+			}
+		}
```

Comment:
I meant do it in `controller.go`, this won't be called on deletions

Reply:
I'd do it in a DeleteFunc on: https://github.com/knative/eventing/blob/66db463e394053ac1bff0556f5a584a6f31a90ec/pkg/reconciler/inmemorychannel/dispatcher/controller.go#L124

Reply:
I think you also want to fetch the object using this helper (then cast it to IMC): https://github.com/knative/pkg/blob/e9570bdb548d85a0ce1d73f64f348771377ae56f/kmeta/accessor.go#L47

Reply:
@vaikas LMK if you wanna sync on this so we can land it 😉

Reply:
I'll try to get this done today, working on moving the rabbitmq into the new test infra.